unixfs/format.go

```diff
@@ -98,32 +105,52 @@ func DataSize(data []byte) (uint64, error) {
 	}
 }
 
-type MultiBlock struct {
+type FSNode struct {
 	Data       []byte
 	blocksizes []uint64
 	subtotal   uint64
+	Type       pb.Data_DataType
 }
 
-func (mb *MultiBlock) AddBlockSize(s uint64) {
+func FSNodeFromBytes(b []byte) (*FSNode, error) {
+	pbn := new(pb.Data)
+	err := proto.Unmarshal(b, pbn)
+	if err != nil {
+		return nil, err
+	}
+
+	mb := new(FSNode)
```

Comment:
`fsn` or maybe just `n` instead? Same for the method receiver var names. `mb` won't make sense in the future.